doc/api/fs.md

```diff
@@ -2069,6 +2107,117 @@ through `fs.open()` or `fs.writeFile()`) will fail with `EPERM`. Existing hidden
 files can be opened for writing with the `r+` flag. A call to `fs.ftruncate()`
 can be used to reset the file contents.
 
+## fs.openFD(path, flags[, mode], callback)
```

Comment:
should this return a `Promise` instead of returning a callback? the rest of the code is promise-based. It should be either 100% callbacks or 100% promises, having a callback here but `fd.close().then()` is not consistent.

Reply:
Yeah, I'm going back and forth on this and thinking that rather than introduce `fs.openFD()` and `fs.openFDSync()` we just introduce a Promise returning `fs.promises.openFD()` and keep this isolated to use of the Promises API for now.

Reply:
And if that is the approach to take on this, then I'll just close this PR and bundle this commit in with the one that introduces the initial broader set of promisified functions